howso/direct/client.py

```diff
@@ -516,56 +513,64 @@ def create_trainee(  # noqa: C901
                 f"binaries/camls or {trainee_id} trainee already exists.")
 
-        metadata = {
-            'name': trainee.name,
-            'metadata': trainee.metadata,
-            'persistence': trainee.persistence,
-        }
+        metadata = dict(
+            name=name,
+            persistence=persistence,
+            **(metadata or {})
+        )
+        new_trainee = dict(
+            name=name,
+            features=features,
+            persistence=persistence,
+            id=id,
+            metadata=metadata
+        )
+        new_trainee = internals.preprocess_trainee(new_trainee)
         self.howso.set_metadata(trainee_id, metadata)
-        self.howso.set_feature_attributes(trainee_id, trainee.features)
-        trainee.features = self.howso.get_feature_attributes(trainee_id)
+        self.howso.set_feature_attributes(trainee_id, new_trainee['features'])
+        new_trainee['features'] = self.howso.get_feature_attributes(trainee_id)
+        new_trainee = internals.postprocess_trainee(new_trainee)
 
         self._output_version_in_trace(trainee_id)
 
-        new_trainee = internals.postprocess_trainee(trainee)
         self.trainee_cache.set(new_trainee)
         return new_trainee
 
-    def update_trainee(self, trainee: Trainee) -> Trainee:
+    def update_trainee(self, trainee: Dict) -> Dict:
         """
         Update an existing Trainee in the Howso service.
 
         Parameters
         ----------
-        trainee : Trainee
+        trainee : Dict
             A `Trainee` object defining the Trainee.
 
         Returns
         -------
-        Trainee
+        Dict
             The `Trainee` object that was updated.
         """
-        if trainee.id:
-            trainee_id = trainee.id
+        if 'id' in trainee:
+            trainee_id = trainee['id']
         else:
-            trainee_id = trainee.id = trainee.name
+            trainee_id = trainee['id'] = trainee["name"]
 
         if not trainee_id:
             raise ValueError("A trainee id is required.")
 
         self._auto_resolve_trainee(trainee_id)
         if self.verbose:
-            print(f'Updating trainee with id: {trainee.id}')
+            print(f'Updating trainee with id: {trainee["id"]}')
 
-        trainee = internals.preprocess_trainee(trainee)
-        metadata = {
-            'name': trainee.name,
-            'metadata': trainee.metadata,
-            'persistence': trainee.persistence,
-        }
+        metadata = dict(
+            name=trainee["name"],
+            metadata=trainee["metadata"],
+            persistence=trainee["persistence"],
+        )
         self.howso.set_metadata(trainee_id, metadata)
-        self.howso.set_feature_attributes(trainee_id, trainee.features)
-        trainee.features = self.howso.get_feature_attributes(trainee_id)
+        trainee["metadata"] = self.howso.get_metadata(trainee_id)
+        self.howso.set_feature_attributes(trainee_id, trainee["features"])
```

Comment:
The trainee user metadata is a subkey of the full metadata that is stored in the trainee. We reserve top level keys for soring other items like the name and persistence (and any future data). While a subkey will map the the user's metadata.
```suggestion
        self.howso.set_metadata(trainee_id, metadata)
        self.howso.set_feature_attributes(trainee_id, trainee["features"])
```

Reply:
This may have been addressed by correction made from your earlier comment:
https://github.com/howsoai/howso-engine-py/pull/205/files/292ee76c8666fa4a46f35b03b0c42929ba2f7f86#r1639107427

Reply:
See my response to linked discussion. This still needs a change.